internal/saucecloud/cloud.go

```diff
@@ -28,15 +28,16 @@ import (
 
 // CloudRunner represents the cloud runner for the Sauce Labs cloud.
 type CloudRunner struct {
-	ProjectUploader storage.ProjectUploader
-	JobStarter      job.Starter
-	JobReader       job.Reader
-	JobWriter       job.Writer
-	JobStopper      job.Stopper
-	CCYReader       concurrency.Reader
-	TunnelService   tunnel.Service
-	Region          region.Region
-	ShowConsoleLog  bool
+	ProjectUploader    storage.ProjectUploader
+	JobStarter         job.Starter
+	JobReader          job.Reader
+	JobWriter          job.Writer
+	JobStopper         job.Stopper
+	CCYReader          concurrency.Reader
+	TunnelService      tunnel.Service
+	Region             region.Region
+	ShowConsoleLog     bool
+	ArtifactDownloader artifact.Downloader
```

Comment:
It's a bit odd that ArtifactDownloader is not an interface 🤔 It'd be OK for now, but it's actually techdebt in the making.
If you look closely, the artifact downloader is nothing but a facade over resto. Resto is the implementation of the artifact downloader. Wouldn't it then make more sense to have an interface that resto implements? That'd make your tests easier too, as you can just mock the artifact downloader, rather than its dependencies :)

Reply:
updating